lib/pg.provider.js

```diff
@@ -530,11 +530,16 @@ class PostgresProvider extends StorageProvider {
   //     err - <Error> | <null>
   //     count - <number>
   delete(category, query, callback) {
-    const categoryDefinition =
-      this.gs.schema.categories.get(category).definition;
-    // TODO: check here if the category is not referenced in another category's
-    // Include decorator after metaschema is updated (required functionality
-    // was added in https://github.com/metarhia/metaschema/pull/170)
+    const categorySchema = this.gs.schema.categories.get(category);
+    const categoryDefinition = categorySchema.definition;
+    if (categorySchema.references.Include.length !== 0) {
+      process.nextTick(callback, new GSError(
+        errorCodes.INVALID_DELETION_OPERATION,
+        `Cannot delete instances of category ${category}, it is included` +
+        ` in categories ${categorySchema.references.Include.join(', ')}`
```

Comment:
Nit: I'm not sure if we should return all of Include categories. This may have both security implications and lead to enormous error messages.
Though this is not an issue right now.

Reply:
@lundibundi this message is not being sent to the client, so I am not sure if there are any security implications. As for enormous error messages, I think, the possible solution may be to report only a part of this list.
https://github.com/metarhia/globalstorage/blob/2b4d6eb7bff65df4a169fb8fa040ee99b3c91ec7/lib/remote.provider.jstp.api.js#L112-L114